tensorflow/compiler/tf2tensorrt/convert/ops/layer_utils.h

```diff
@@ -0,0 +1,354 @@
+
+  // Adds an Add operation to the network.
+  StatusOr<nvinfer1::IElementWiseLayer*> Sub(nvinfer1::ITensor* lhs,
+                                             nvinfer1::ITensor* rhs) noexcept {
+    TRT_EXPECT(network_);
+    TRT_EXPECT(lhs);
+    TRT_EXPECT(rhs);
+    auto layer = network_->addElementWise(*lhs, *rhs,
+                                          nvinfer1::ElementWiseOperation::kSUB);
+    return MakeResult(layer);
+  }
+
+  // Adds an Greater operation to the network.
+  StatusOr<nvinfer1::IElementWiseLayer*> Greater(
+      nvinfer1::ITensor* lhs, nvinfer1::ITensor* rhs) noexcept {
+    TRT_EXPECT(network_);
+    TRT_EXPECT(lhs);
+    TRT_EXPECT(rhs);
+    auto layer = network_->addElementWise(
+        *lhs, *rhs, nvinfer1::ElementWiseOperation::kGREATER);
+    return MakeResult(layer);
+  }
+
+  // Adds an Equal operation to the network.
+  StatusOr<nvinfer1::IElementWiseLayer*> Equal(
+      nvinfer1::ITensor* lhs, nvinfer1::ITensor* rhs) noexcept {
+    TRT_EXPECT(network_);
+    TRT_EXPECT(lhs);
+    TRT_EXPECT(rhs);
+    auto layer = network_->addElementWise(
+        *lhs, *rhs, nvinfer1::ElementWiseOperation::kEQUAL);
+    return MakeResult(layer);
+  }
+
+  // Simulates the creation of a tensor of the same shape as "input", but filled
+  // with 0's where "input" is non-zero. TensorRT doesn't have dynamic creation
+  // functions, so this function multiplies the input by a zero value to side
+  // step knowing the shape. For static inputs, this can be optimized using one
+  // of the TRTNetworkBuilder::Constant function instead.
+  StatusOr<nvinfer1::IElementWiseLayer*> ZerosLikeDyamic(
+      nvinfer1::ITensor* input) noexcept {
+    TRT_EXPECT(network_);
+    TRT_EXPECT(input);
+    // TODO: add switch statement
+    auto zero = this->Constant<int32>(0, input->getDimensions().nbDims);
+    TRT_ENSURE_OK(zero);
+    return Mul(input, (*zero)->getOutput(0));
+  }
+
+  // Adds an Add operation to the network.
+  StatusOr<nvinfer1::IElementWiseLayer*> FloorDiv(
+      nvinfer1::ITensor* lhs, nvinfer1::ITensor* rhs) noexcept {
+    TRT_EXPECT(network_);
+    TRT_EXPECT(lhs);
+    TRT_EXPECT(rhs);
+    auto layer = network_->addElementWise(
+        *lhs, *rhs, nvinfer1::ElementWiseOperation::kFLOOR_DIV);
+    return MakeResult(layer);
+  }
+
+  // Abs(Ceiling divide(x/y)) operation for integers
+  StatusOr<nvinfer1::IElementWiseLayer*> AbsCeilDivInt(
+      nvinfer1::ITensor* lhs, nvinfer1::ITensor* rhs) noexcept {
+    TRT_EXPECT(network_);
+    TRT_EXPECT(lhs);
+    TRT_EXPECT(rhs);
+    auto rhs_abs = this->AbsInt(rhs);
+    TRT_ENSURE_OK(rhs_abs);
+    auto lhs_abs = this->AbsInt(lhs);
+    TRT_ENSURE_OK(lhs_abs);
+    auto add1 = this->Add((*lhs_abs)->getOutput(0), (*rhs_abs)->getOutput(0));
+    TRT_ENSURE_OK(add1);
+    auto one_const = this->Constant<int32>(1, rhs->getDimensions().nbDims);
+    TRT_ENSURE_OK(one_const);
+    auto numerator =
+        this->Sub((*add1)->getOutput(0), (*one_const)->getOutput(0));
```

Comment:
We shouldn't use these "auto"s here, according to the style guide. Right?
There are similar cases in this file.

Reply:
fixed